tests/test_addons_toupdate.py

```diff
@@ -0,0 +1,610 @@
+    with working_directory(initdir):
+        res = runner.invoke(main, [
+            'addons',
+            'toupdate',
+            '15150ba2150bd4b796b80a82972625440c1b4820',
+            '-r',
+            '--requirement',
+            './requirements.txt',
+        ])
+        assert 0 == res.exit_code
+        expected = '\n'
+        assert expected == res.output
+
+
+def test_addon_toupdate_16(initdir):
+    """
+    Data:
+        - addon_1 exists
+        - addon_2 exists
+        - requirements file exists
+    Test case:
+        - changes in addon_2
+        - compare requirements: YES
+    Expected result:
+        - changed addon's name (addon_2)
+    """
+    runner = CliRunner()
+    _checkout_to(initdir, '1c0e49e58f6a9b36c14becbbdb039cbca36f5293')
+    with working_directory(initdir):
+        res = runner.invoke(main, [
+            'addons',
+            'toupdate',
+            'a01139afa02021954c0c9ffd899c794bb923fef9',
+            '-r',
+            '--requirement',
+            './requirements.txt',
+        ])
+        assert 0 == res.exit_code
+        expected = 'addon_2\n'
+        assert expected == res.output
+
+
+def test_addon_toupdate_17(initdir):
+    """
+    Data:
+        - addon_1 exists
+        - addon_2 exists
+        - requirements file exists
+    Test case:
+        - remove external odoo addons source
+        - compare requirements: YES
+    Expected result:
+        - /
+    """
+    runner = CliRunner()
+    _checkout_to(initdir, '3382b4e8afb086b0aaa4946f9bf9265c7a41c768')
+    with working_directory(initdir):
+        res = runner.invoke(main, [
+            'addons',
+            'toupdate',
+            '1c0e49e58f6a9b36c14becbbdb039cbca36f5293',
+            '-r',
+            '--requirement',
+            './requirements.txt',
+        ])
+        assert 0 == res.exit_code
+        expected = '\n'
+        assert expected == res.output
+
+
+def test_addon_toupdate_18(initdir):
+    """
+    Data:
+        - addon_1 exists
+        - addon_2 exists
+        - requirements file exists
+    Test case:
+        - remove odoo addon dependency
+        - compare requirements: YES
+    Expected result:
+        - /
+    """
+    runner = CliRunner()
+    _checkout_to(initdir, 'bfde22f2c292ecc2921e1b7fc09b31bdf79b1f82')
+    with working_directory(initdir):
+        res = runner.invoke(main, [
+            'addons',
+            'toupdate',
+            '3382b4e8afb086b0aaa4946f9bf9265c7a41c768',
+            '-r',
+            '--requirement',
+            './requirements.txt',
+        ])
+        assert 0 == res.exit_code
+        expected = '\n'
+        assert expected == res.output
+
+
+def test_addon_toupdate_19(initdir):
+    """
+    Data:
+        - addon_1 exists
+        - addon_2 exists
+        - requirements file exists
+    Test case:
+        - remove library dependency
+        - compare requirements: YES
+    Expected result:
+        - /
+    """
```

Comment:
What would be nice for readbility of the test docstring is to add a link such as
https://github.com/acsone/acsoo/compare/bfde22f...657f0f9
That could even replace the Data/Test case part.

That said, it's strange that the diff is empty here.